File: computer-selling-v1/src/app/model/response/promotion.response.ts
import { ProductItemResponse } from "./product.response"

export interface Promotion {
    promotionId: number
    name: string
    description: string
    discountPercent: number
    discountAmount: number
    startDate: string
    endDate: string
    status: string
    product: ProductItemResponse
}
export interface PromotionById {
    promotionId: number
    name: string
    description: string
    discountPercent: number
    discountAmount: number
    startDate: string
    endDate: string
    status: string
}

File: identity-service/src/main/java/com/example/identity_service/dto/response/PromotionResponse.java
package com.example.identity_service.dto.response;

import com.example.identity_service.enums.PromotionStatus;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.math.BigDecimal;
import java.time.LocalDate;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class PromotionResponse {
     Long promotionId;

     String name; // Tên chương trình giảm giá

     String description; // Mô tả chương trình

     BigDecimal discountPercent; // % giảm giá (nếu có)

     BigDecimal discountAmount; // Hoặc giảm số tiền cố định

     LocalDate startDate; // Ngày bắt đầu

     LocalDate endDate; // Ngày kết thúc

     PromotionStatus status; // ACTIVE, INACTIVE, CANCELLED
}
